sys/include/ng_net/ng_nettype.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net
+ * @{
+ *
+ * @file
+ * @brief   Protocol type definitions
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NG_NETTYPE_H_
+#define NG_NETTYPE_H_
+
+/**
+ * @brief   Definition of protocol types in the network stack.
+ *
+ * @note    Expand at will.
+ */
+typedef struct {
+    NG_NETTYPE_UNDEF = 0,       /**< Protocol is undefined */
+    /**
+     * @{
+     * @name Link Layer
+     */
+#ifdef MODULE_NG_IEEE802154
+    NG_NETTYPE_IEEE802154,      /**< Protocol is IEEE 802.15.4 */
+#endif
+#ifdef MODULE_NG_SIXLOWPAN
+    NG_NETTYPE_SIXLOWPAN,       /**< Protocol is 6LoWPAN */
+#endif
+    /**
+     * @}
+     */
+
+    /**
+     * @{
+     * @name Network layer
+     */
+#ifdef MODULE_NG_IPV6
+    NG_NETTYPE_IPV6,            /**< Protocol is IPv6 */
+#endif
+#ifdef MODULE_NG_ICMPV6
+    NG_NETTYPE_ICMPV6,          /**< Protocol is ICMPv6 */
+    NG_NETTYPE_ICMPV6_ECHO,     /**< Protocol is ICMPv6, echo reply/request */
```

Comment:
Didn't you complain today if someone would want to register for ICMPv6 echo requests/replies they would get all ICMPv6 packets and would need to always be woken up if one is received even though it might not be an echo request/reply?